Core/OxidEeEvents.php

```diff
@@ -382,6 +403,40 @@ private static function _addPaymentMethod($oPayment)
         );
     }
 
+    /**
+     * Adds initial SEPA mandate text to SEPA Direct Debit payment method
+     *
+     * @since 1.1.0
+     */
+    private static function _insertSepaMandate()
+    {
+        $sSepaMandate = self::_prepareSepaMandate(0);
+        $sSepaMandate1 = self::_prepareSepaMandate(1);
+        $sQuery = "UPDATE oxpayments SET `WDOXIDEE_SEPAMANDATECUSTOM` = '$sSepaMandate', `WDOXIDEE_SEPAMANDATECUSTOM_1`
+            = '$sSepaMandate1' WHERE `OXID` LIKE " . "'" . self::SEPA_DD_PAYMENT_ID . "'";
+        self::$oDb->execute($sQuery);
+    }
+
+    /**
+     * Prepares SEPA mandate text for the given language id
+     *
+     * @param integer $iLanguageId
+     *
+     * @return string
+     *
+     * @since 1.1.0
+     */
+    private static function _prepareSepaMandate($iLanguageId)
+    {
+        $oLanguage = oxNew(Language::class);
+        return $oLanguage->translateString('wd_sepa_text_1', $iLanguageId) . ' ' . '%creditorName%' .
```

Comment:
I think we should extend `Helper::translate()` to also support passing a language ID. Then you can use it here as well.